### PR TITLE
lexicon: 3.9.4 -> 3.11.2

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

We could actually move lexicon to `pythonPackages` as dns-lexicon since it's also a Python library: https://github.com/AnalogJ/lexicon#id1

@blitz @flyfloh You should step up and maintain softlayer as well as softlayer-zeep.

### CODE REVIEW

**Review by JohnAZoidberg — CHANGES_REQUESTED** [review]
The tests don't pass: https://github.com/NixOS/nixpkgs/pull/176287/checks?check_run_id=6740652896

Confirmed locally with `nixpkgs-review pr 176287`:

```
error: builder for '/nix/store/bkh4hjxa5babhm7zrnw475srja7yp0y2-lexicon-3.11.2.drv' failed with exit code 1;
       last 10 log lines:
       > >       with io.open(filename, mode="rb") as f:
       > E       FileNotFoundError: [Errno 2] No such file or directory: 'oci_api_key.pem'
       >
       > /nix/store/q6mxvqf6j00sl0qyy8vjsm5v1pji0gx7-python3.9-oci-2.63.0/lib/python3.9/site-packages/oci/signer.py:31: FileNotFoundError
       > =========================== short test summary info ============================
       > FAILED lexicon/tests/providers/test_oci.py::OciProviderTests::test_provider_when_calling_create_record_for_A_with_valid_name_and_content
       > FAILED lexicon/tests/providers/test_oci.py::OciProviderTests::test_provider_when_calling_list_records_after_setting_ttl
       > FAILED lexicon/tests/providers/test_oci.py::OciProviderTests::test_provider_when_calling_create_record_for_TXT_with_fqdn_name_and_content
       > FAILED lexicon/tests/providers/test_oci.py::OciProviderTests::test_provider_when_calling_list_records_with_fqdn_name_filter_should_return_record
       > =========== 4 failed, 2019 passed, 217 skipped in 129.41s (0:02:09) ============
       For full logs, run 'nix log /nix/store/bkh4hjxa5babhm7zrnw475srja7yp0y2-lexicon-3.11.2.drv'.
error: 1 dependencies of derivation '/nix/store/m7gaizpmxqf5mhmd8fv8zshyqx28ki52-review-shell.drv' failed to build
```

**Comment by dotlambda** [comment]
> The tests don't pass: https://github.com/NixOS/nixpkgs/pull/176287/checks?check_run_id=6740652896

It's a race condition, probably because multiple tests are run in parallel.
Someone should open an upstream issue. I'll do so in a few days if I find the time.

**Comment by blitz** [comment]
> > The tests don't pass: https://github.com/NixOS/nixpkgs/pull/176287/checks?check_run_id=6740652896
> 
> It's a race condition, probably because multiple tests are run in parallel. Someone should open an upstream issue. I'll do so in a few days if I find the time.

Build and test is fine for me. I don't have a good way to test this right now, but maybe @flyfloh can give this a shot.

File: pkgs/development/python-modules/softlayer-zeep/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    export HOME=$TEMP
+  '';
+
+  pythonImportsCheck = [ "zeep" ];
+
+  meta = {
+    description = "A modern/fast Python SOAP client based on lxml / requests";
+    homepage = "https://github.com/softlayer/softlayer-zeep";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/softlayer-zeep/default.nix:66]:
```suggestion
    maintainers = with lib.maintainers; [ dotlambda ];
```

File: pkgs/development/python-modules/softlayer/default.nix [hunk 2]
```diff
@@ -3,54 +3,65 @@
...
   pythonImportsCheck = [ "SoftLayer" ];
 
   meta = with lib; {
-    broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
+    broken = stdenv.isDarwin;
     description = "Python libraries that assist in calling the SoftLayer API";
     homepage = "https://github.com/softlayer/softlayer-python";
+    changelog = "https://github.com/softlayer/softlayer-python/blob/${src.rev}/CHANGELOG.md";
     license = licenses.mit;
     maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/softlayer/default.nix:66]: Please consider adding yourself here.

File: pkgs/development/python-modules/softlayer-zeep/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/softlayer-zeep/default.nix:66]: > @blitz @flyfloh You should step up and maintain softlayer as well as softlayer-zeep.

**Comment by dotlambda** [comment]
@blitz @flyfloh If you don't want to maintain those Python packages then we should open a PR removing lexicon instead.

**Comment by dotlambda** [comment]
@blitz Alright, I'll maintain the softlayer Python package. But can I add you as maintainer of lexicon?
Also, would you be willing to open an upstream issue about the race condition in `lexicon/tests/providers/test_oci.py`?

**Comment by wegank** [comment]
lexicon is at 3.16.1 on master, softlayer on 6.1.11 and upstream doesn't depend on softlayer-zeep anymore. Closing.

**Pull request closed without merging**